[issue 1397] - Adding support to mount volumes with free string configuration option

fabiobrz opened the pull request:
#### Short description of what this resolves:
Changes allow to set a free form bind mount configuration option when creating a docker container so that - for example - `:Z` can be set for container that run on SELinux.

#### Changes proposed in this pull request:

- Adding an overloaded version of `withVolume(String, String, String)` to ContainerBuilder, ContainerDsl and ContainerDslRule to allow setting free form option for volume mounts.

Fixes #1397

gaol commented on docker/ftest-docker-junit-rule/src/test/java/org/arquillian/cube/docker/junit/rule/PostgreSqlIT.java:
```diff
@@ -0,0 +1,99 @@
+        // This is needed to run on SELinux enabled Docker hosts
+        // otherwise the following error will prevent the container from starting successfully:
+        //   mkdir: cannot create directory '/var/lib/pgsql/data/userdata': Permission denied
+        java.nio.file.Files.setPosixFilePermissions(Path.of(postgresDataDir.toURI()),
```
Shall we check the OS before calling `setPosixFilePermissions` in case running on Windows ?

fabiobrz replied:
Thanks, fixed.

gaol commented on docker/ftest-docker-containerobject-dsl/src/test/java/org/arquillian/cube/containerobject/dsl/PostgreSqlIT.java:
```diff
@@ -0,0 +1,97 @@
+        // This is needed to run on SELinux enabled Docker hosts
+        // otherwise the following error will prevent the container from starting successfully:
+        //   mkdir: cannot create directory '/var/lib/pgsql/data/userdata': Permission denied
+        java.nio.file.Files.setPosixFilePermissions(Path.of(POSTGRESQL_HOST_DATA_DIRECTORY.toURI()),
```
Shall we check the OS before calling `setPosixFilePermissions` in case running on Windows  ?

fabiobrz replied:
Thanks, fixed.

gaol commented on docker/ftest-docker-junit-rule/src/test/java/org/arquillian/cube/docker/junit/rule/PostgreSqlIT.java:
```diff
@@ -0,0 +1,99 @@
+ *     {@code mkdir: cannot create directory '/var/lib/pgsql/data/userdata': Permission denied}
+ * </p>
+ */
+@Category({ RequiresDocker.class})
```
Shall we add `@RunWith(ArquillianConditionalRunner.class)` to make the `RequiresDocker` work as expected or am I missing something ?

fabiobrz replied:
It is not always needed, as in this case, i.e. when using `@ClassRule` with `ContainerDslRule', see https://arquillian.org/arquillian-cube/#_junit_rule

gaol replied:
Ah, yes. that is the thing I am missing :) thank you .

gaol reviewed (commented):
Overall it looks good to me :+1: 

I have some comments though on the test codes just in case my concerns are valid :)

fabiobrz commented:
> Overall it looks good to me 👍
> 
> I have some comments though on the test codes just in case my concerns are valid :)

Thanks for your review @gaol, I've addressed your comments. Feel free to let me know if there's anything else you'd wish to discuss.
